CONTRIBUTING.md

```diff
@@ -1,27 +1,35 @@
-# Contributing a Devfile Stack to this registry
+# Contributing to this registry
 
-This document outlines the requirements for contributing a devfile stack to this repository
+This document outlines the requirements for contributing a devfile stack or sample to this repository
 
 The [devfile registry structure](https://github.com/devfile/api/blob/main/docs/proposals/registry/registry-structure.md#repository-structure) design doc provides some useful background information on the structure of resources in a devfile registry (and its Git repository).
 
 ## Prerequisites
 
-The following are required to build the devfile index container image containing your stack:
+The following are required to build the devfile index container image containing your stack or sample:
 
-- Golang 1.13.x or later
 - Docker 17.06 or later
 - Git
 
-## Steps
+## Stacks
+
+### Contributing
 
 1) Verify your Devfile stack functions with odo.
   
     - Core odo functions such as `odo create --devfile <devfile.yaml>`, `odo push`, `odo url create` should work with the devfile.
     - PR tests on this repository will verify this functionality as well.
 
+2) Verify your Devfile stack has the following metadata fields at a minimum:
+
+    - Name: The name of the devfile stack, e.g. `java-springboot`
+    - Display Name: The longer name of your devfile stack, e.g. `Spring Boot®`
+    - Description: A brief description of your devfile stack, e.g. `Spring Boot® using Java`
+    - Version: The version of your stack, in semnatic version format, e.g. `1.0.0`
+
 2) Add a folder for the stack to `stacks/` in this repository.
   
-    - Name format should be in the form of `<language>-<framework>`. E.g. `java-quarkus`, `python-django`, etc.
+    - The name should match the devfile stack's name, and be in the format `<language>-<framework>`. E.g. `java-quarkus`, `python-django`, etc.
```

Comment:
_Should_ the name match the devfile stack? Or _must_ it? If the latter, simply say, `The name must match the devfile stack's name` or make it user focused like, `Make sure the name matches the devfile stack's name and be in the format...`